### ISSUE TITLE
Add different tag_ids in animals

### ISSUE THREAD

**Issue by peterdesmet** [inbo/etn#108]
@PieterjanVerhelst:

> I think animal_nickname will be used; for twaite shad we did this, but I don't think it was fed into the database.
> 
> @IPauwels, correct remark about double tagging. I prefer various tag information is given in the same animal/line. Would it be an option to make a distinction by different tag IDs, by including the telemetry type in the header's name:
> - acoustic_tag_id
> - pit_tag_id
> - radio_tag_id
> - dst_tag_id
> - psat_tag_id
> 
> These are the most common applied telemetry techniques.
> 
> Another note: on the ETN database you can switch between different telemetry techniques on the top left of the website; now you have the choice between acoustic and c-pods. Hence, the acoustic_tag_id field should be the default and only mandatory field.
@jreubens , what is your opinion on this?
> 
> _Originally posted by @PieterjanVerhelst in https://github.com/inbo/etn/issues/74#issuecomment-601051894_

**Comment by peterdesmet** [comment]
https://github.com/inbo/etn/issues/74#issuecomment-601080633 (regarding _using_ other tags):

> I myself don't at the moment. However, I have eels in the database with both an acoustic and pop-off tag, but only the acoustic tag is indicated because there is no field yet for the pop-off tag ID. @IPauwels did you store PIT IDs in the database? I would assume this is added to the `comments` field of the `animal` information

**Comment by jreubens** [comment]
the choice between 'cpod' and 'acoustic telemetry' should be renamed to 'telemetry' in general. This covers all types of telemetry.
You specify your 'type' thus no need to have this mentioned before ID to my opinion

**Comment by jreubens** [comment]
there is no metadata on other telemetry types yet in the DB. It still needs to be developed.
The only thing that is already possible is to upload project info on other telemetry types

**Comment by PieterjanVerhelst** [comment]
But it would be interesting to add different tag IDs from different telemetry types to an animal.

**Comment by IPauwels** [comment]
> But it would be interesting to add different tag IDs from different telemetry types to an animal.

Yes indeed, I agree.

**Comment by IPauwels** [comment]
> [#74 (comment)](https://github.com/inbo/etn/issues/74#issuecomment-601080633) (regarding _using_ other tags):
> 
> > I myself don't at the moment. However, I have eels in the database with both an acoustic and pop-off tag, but only the acoustic tag is indicated because there is no field yet for the pop-off tag ID. @IPauwels did you store PIT IDs in the database? I would assume this is added to the `comments` field of the `animal` information

No, we don't have PIT data in the database yet, as Jan also indicated already that only project info on other telemetry data can be added so far, but not meta and detection data.

We don't have double tagged animals either, so no PIT label info in the database yet from INBO.

**Comment by peterdesmet** [comment]
> But it would be interesting to add different tag IDs from different telemetry types to an animal.

Agree, but then we should maybe use the same strategy as currently: by having one row per animal-tag, with the tag (of any type) in `tag_id`, rather than adding more columns, which isn't scalable for tags of the same type anyway. Right now, the R function is going to return that information collapsed: one row per animal, with tag identifiers comma separated.

**Comment by IPauwels** [comment]
> Right now, the R function is going to return that information collapsed: one row per animal, with tag identifiers comma separated.

I think this is a perfect way of sticking to the same strategy and still being able to get the info returned in one row (in R then, not in the database).
So I would say: let's stick to the same strategy and having a double tagged animal twice in the database (two separate rows) with another tag per row..

**Comment by peterdesmet** [comment]
> get the info returned in one row (in R then, not in the database)

Note that it is also returned similarly in the etn application, so that is consistent as well:

![](https://user-images.githubusercontent.com/600993/74852124-78ea8500-533c-11ea-8bda-94651cb1a510.png)

**Comment by peterdesmet** [comment]
If we go this route, it is probably useful to also include the tag type in the animals view, so you can immediately see (comma-separated) what tag types are added to an animal.

I assume that would the `telemetry_type` field (from the `tags` view), which currently has the following values: `acoustic`, `Acoustic`, `acoustic tag`, `ADST` (ideally more controlled, and where PIT etc. would be additional types) and not `type`, which has the values `animal`? Can anyone conform this?

**Comment by PieterjanVerhelst** [comment]
Indeed, I agree. `acoustic` can be used for acoustic telemetry type.

**Comment by fwaumans** [comment]
Now, you can only store acoustic tags in the database.   In the tag table one record is actually one transmitter. An acoustic tag with two transmitters (ex. P and T) is stored as two records in the tag table with the same serial number in the serial number field.  From ADST tags, only  the acoustic transmitter can be stored in the database.  We plan to extend the database to store DST.
